File: go/algoritms/sliding.go
package algoritms

import "math"

func MinSubArrayLenSliding(arr []int, target int) int {
	n := len(arr)
	left := 0
	sum := 0
	minLen := math.MaxInt32

	for right := 0; right < n; right++ {
		sum += arr[right]

		for sum >= target {
			if right-left+1 < minLen {
				minLen = right - left + 1
			}
			sum -= arr[left]
			left++
		}
	}

	if minLen == math.MaxInt32 {
		return 0
	}
	return minLen
}

File: cpp/algorithms/sliding.cpp
#include "algorithms.hpp"
#include <climits>

using namespace std;

// Sliding window implementation to find the minimal length of a contiguous subarray
// of which the sum is at least target. Returns 0 if no such subarray exists.
// Time Complexity: O(n)
int minSubArrayLenSliding(const vector<int>& nums, int target) {
  int n = nums.size();
  int left = 0, sum = 0, minLength = INT_MAX;

  for (int right = 0; right < n; ++right) {
      sum += nums[right];

      while (sum >= target) {
          minLength = min(minLength, right - left + 1);
          sum -= nums[left];
          ++left;
      }
  }

  return (minLength == INT_MAX) ? 0 : minLength;
}